src/app/shared/links/links-list/links-list-table/links-list-table.component.ts

```diff
@@ -106,21 +117,69 @@ export class LinksListTableComponent implements OnChanges {
     }, []);
   }
 
-  public documentSelected(collection: CollectionModel, linkRow: LinkRowModel) {
-    const document = linkRow.document;
-    this.select.emit({collection, document});
+  private handleNewLinkRows(linkRows: LinkRowModel[]) {
+    if (this.lastSelection && this.lastSelection.document === this.document && this.lastSelection.linkType === this.linkType) {
+      this.mergeLinkRows(linkRows);
+    } else {
+      this.lastSelection = {linkType: this.linkType, document: this.document};
+      this.linkRows$.next(linkRows);
+    }
+  }
+
+  private mergeLinkRows(linkRows: LinkRowModel[]) {
+    const currentLinkRows = this.linkRows$.getValue();
+    const createdLinkRows = currentLinkRows.filter(linkRow => linkRow.document && linkRow.linkInstance);
+    let emptyLinkRows = currentLinkRows.filter(linkRow => linkRow.correlationId);
+
+    if (linkRows.length > createdLinkRows.length && emptyLinkRows.length > 0) {
+      emptyLinkRows = emptyLinkRows.slice(1);
+    }
+
+    const newLinkRows = linkRows.concat(emptyLinkRows);
+    this.linkRows$.next(newLinkRows);
+  }
+
+  public createNewLinkedDocument() {
+    const currentLinkRows = this.linkRows$.getValue();
+    const correlationId = CorrelationIdGenerator.generate();
+    this.linkRows$.next(currentLinkRows.concat([{correlationId}]));
+
+    this.goToLastPage(currentLinkRows.length);
+    this.scrollToNewLinkRow(correlationId);
   }
 
-  public unlinkDocument(linkRow: LinkRowModel) {
-    this.unlink.emit(linkRow.linkInstance.id);
+  private goToLastPage(sizeBeforeAdd: number) {
+    this.page = Math.floor(sizeBeforeAdd / this.pageSize);
   }
 
-  public trackByAttribute(index: number, attribute: AttributeModel): string {
-    return attribute.correlationId || attribute.id;
+  private scrollToNewLinkRow(correlationId: string) {
+    setTimeout(() => {
+      const element = document.getElementById(correlationId);
+      if (element) {
+        const scrollableParent = element.parentElement.parentElement.parentElement;
+        if (scrollableParent) {
+          scrollableParent.scrollTop = Number.MAX_SAFE_INTEGER;
```

Comment:
What about using [Element.scrollIntoView()](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) instead of this complicated structure?

Reply:
Ok